credentials/tls/certprovider/pemfile/builder.go

```diff
@@ -62,32 +63,23 @@ func pluginConfigFromJSON(jd json.RawMessage) (Options, error) {
 	// The only difference between this anonymous struct and the Options struct
 	// is that the refresh_interval is represented here as a duration proto,
 	// while in the latter a time.Duration is used.
-	cfg := &struct {
-		CertificateFile          string          `json:"certificate_file,omitempty"`
-		PrivateKeyFile           string          `json:"private_key_file,omitempty"`
-		CACertificateFile        string          `json:"ca_certificate_file,omitempty"`
-		SPIFFETrustBundleMapFile string          `json:"spiffe_trust_bundle_map_file,omitempty"`
-		RefreshInterval          json.RawMessage `json:"refresh_interval,omitempty"`
-	}{}
-	if err := json.Unmarshal(jd, cfg); err != nil {
+	opts := Options{
+		RefreshDuration: defaultRefreshInterval,
+	}
+
+	if err := json.Unmarshal(jd, &opts); err != nil {
 		return Options{}, fmt.Errorf("pemfile: json.Unmarshal(%s) failed: %v", string(jd), err)
 	}
 
-	opts := Options{
-		CertFile:            cfg.CertificateFile,
-		KeyFile:             cfg.PrivateKeyFile,
-		RootFile:            cfg.CACertificateFile,
-		SPIFFEBundleMapFile: cfg.SPIFFETrustBundleMapFile,
-		// Refresh interval is the only field in the configuration for which we
-		// support a default value. We cannot possibly have valid defaults for
-		// file paths to watch. Also, it is valid to specify an empty path for
-		// some of those fields if the user does not want to watch them.
-		RefreshDuration: defaultRefreshInterval,
+	var rawMap map[string]json.RawMessage
```

Comment:
I think this isn't quite what we want to do - here we've just swapped the `Options` struct for a map without solving the default issue

I think the idea here is rather that we don't need two structs (or most of this code) at all, so swapping one for a map doesn't do it - rather, we should be able to handle a default `RefreshDuration` during the unmarshal call without having an intermediate in the first place.

Reply:
it seems like Options struct needs RefreshInterval field for unmarshalling.